et/cs-229-machine-learning-tips-and-tricks.md

```diff
@@ -0,0 +1,289 @@
+
+<br>
+
+**24. [Training set, Validation set, Testing set]**
+
+&#10230; [Treenimiskomplekt, Valideerimiskomplekt, Testimiskomplekt]
+
+<br>
+
+**25. [Model is trained, Model is assessed, Model gives predictions]**
+
+&#10230; [Mudel treenitakse, Mudel valideeritakse, Mudel annab ennustusi]
+
+<br>
+
+**26. [Usually 80% of the dataset, Usually 20% of the dataset]**
+
+&#10230; [Tavaliselt 80% andmekogust, Tavaliselt 20% andmekogust]
+
+<br>
+
+**27. [Also called hold-out or development set, Unseen data]**
+
+&#10230; [Nimetatakse arenduskomplektiks, andmed mida mudel pole näinud]
+
+<br>
+
+**28. Once the model has been chosen, it is trained on the entire dataset and tested on the unseen test set. These are represented in the figure below:**
+
+&#10230; Kui mudel on valitud, treenitakse seda kogu andmehulgaga ja testitakse mitte nähtud andmetega. Need on esitatud alloleval joonisel.
+
+<br>
+
+**29. Cross-validation ― Cross-validation, also noted CV, is a method that is used to select a model that does not rely too much on the initial training set. The different types are summed up in the table below:**
+
+&#10230; Ristvalideerimine - Ristvalideerimine, tuntud kui CV, on meetod, mida kasutatakse mudeli valimiseks ning mis ei sõltu liiga palju algsest treeningkomplektist. Erinevad ristkontrolli tüübid on kokkuvõetud allolevas tabelis:
+
+<br>
+
+**30. [Training on k−1 folds and assessment on the remaining one, Training on n−p observations and assessment on the p remaining ones]**
+
+&#10230; [Treenitakse k-1 ühikuga (treeningkomplektist) ning hindamine ülejäänuga, Treenimine n-p vaatlusega ja hindamine ülejäänud p vaatlustega]
+
+<br>
+
+**31. [Generally k=5 or 10, Case p=1 is called leave-one-out]**
+
+&#10230; [Tavaliselt k=5 või 10, Kui p=1 siis selle nimi on jäta-üks-välja]
+
+<br>
+
+**32. The most commonly used method is called k-fold cross-validation and splits the training data into k folds to validate the model on one fold while training the model on the k−1 other folds, all of this k times. The error is then averaged over the k folds and is named cross-validation error.**
+
+&#10230; Kõige sagedamini kasutatavat meetodit nimetatakse k-kordseks ristvalideerimiseks ja see jagab treeningu andmehulga k kordseks osadeks, et mudelit valideerida ühel osaga, samal ajal treenides mudelit k − 1 teisel osaga, seda kõike k korda. Seejärel arvutatakse viga keskmiselt k korda ja seda nimetatakse ristvalideerimise veaks.
+
+<br>
+
+**33. Regularization ― The regularization procedure aims at avoiding the model to overfit the data and thus deals with high variance issues. The following table sums up the different types of commonly used regularization techniques:**
+
+&#10230; Regulariseerimine - Regulariseerimisprotsessi eesmärk on vältida mudeli ülemäärast ülesobitust treeningandmestikule ning kõrge dispersiooni probleemidega. Järgmises tabelis on toodud levinumate reguleerimist meetodite tüübid:
```

Comment:
Instead of ".....treeningandmestikule ning kõrge dispersiooni probleemidega." consider following phrasing:

"treeningandmestikule ning seeläbi tegeleb kõrge dispersiooni probleemidega."